Sources/OpenAPIURLSession/URLSessionTransport.swift

```diff
@@ -73,8 +73,21 @@ public struct URLSessionTransport: ClientTransport {
 
         /// Creates a new configuration with the provided session.
         /// - Parameter session: The URLSession used for performing HTTP operations.
-        ///     If none is provided, the system uses the shared URLSession.
-        public init(session: URLSession = .shared) { self.session = session }
+        ///   If none is provided, the system uses the shared URLSession.
+        public init(session: URLSession = .shared) { self.init(session: session, implementation: .platformDefault) }
+
+        enum Implementation {
+            case buffered
+            case streaming(requestBodyStreamBufferSize: Int, responseBodyStreamWatermarks: (low: Int, high: Int))
+        }
+
+        var implemenation: Implementation
+
+        init(session: URLSession = .shared, implementation: Implementation = .platformDefault) {
```

Comment:
```suggestion
        init(session: URLSession = .shared, implementation: Implementation) {
```

This way `.init()` won't be ambiguous.

Reply:
I'm not seeing any ambiguity issues when calling `URLSessionTransport.Configuration.init()` in the tests, even when `@testable import OpenAPIURLSession`. Note that this initializer is `internal`, if that makes any difference to the ambiguity story.

Reply:
Not sure, but if it's compiling fine, then we can keep it. I'll have to catch up on the precedence rules in Swift, because I'm not sure which one will actually be called in practice.